File: ios/FreelanceKar/Views/Sheets/QuoteSheet.swift
import SwiftUI

struct QuoteSheetView: View {
    @EnvironmentObject var store: AppStore

    var body: some View {
        let s = store.state
        let req = s.reqs.first { $0.id == s.quoteFor }

        FKSheet(isOpen: req != nil, onClose: { store.closeQuote() }, maxHeightFraction: 0.88) {
            if let req {
                VStack(alignment: .leading, spacing: 0) {
                    HStack {
                        Text("Send a quote").font(.fkSheetTitle)
                        Spacer()
                        Button { store.closeQuote() } label: { Image(systemName: "xmark").font(.system(size: 14)).foregroundStyle(FK.text50) }
                    }

                    VStack(alignment: .leading, spacing: 4) {
                        Text(req.title).font(.fkCardTitle)
                        Text(req.meta).font(.system(size: 11.5)).foregroundStyle(FK.text50)
                        Text("Client budget: \(req.budget)").font(.system(size: 11.5, weight: .medium)).foregroundStyle(FK.lime)
                    }
                    .padding(13)
                    .background(FK.nested)
                    .clipShape(RoundedRectangle(cornerRadius: 14, style: .continuous))
                    .overlay(RoundedRectangle(cornerRadius: 14).stroke(FK.line09, lineWidth: 1))
                    .padding(.top, 14)

                    VStack(alignment: .leading, spacing: 8) {
                        MonoLabel(text: "YOUR PRICE")
                        HStack(spacing: 9) {
                            Text("₹").font(.system(size: 14, weight: .medium)).foregroundStyle(FK.text50)
                            TextField("", text: Binding(get: { s.quoteAmount }, set: { store.setQuoteAmount($0) }),
                                      prompt: Text("14000").foregroundStyle(FK.text42))
                                .keyboardType(.numberPad)
                                .font(.system(size: 15, weight: .semibold))
                                .foregroundStyle(FK.text)
                        }
                        .padding(13)
                        .background(FK.nested)
                        .clipShape(RoundedRectangle(cornerRadius: 13, style: .continuous))
                        .overlay(RoundedRectangle(cornerRadius: 13).stroke(FK.line09, lineWidth: 1))
                        Text("You keep 100% — client pays you directly, no platform cut.")
                            .font(.system(size: 11)).foregroundStyle(FK.text42)
                    }.padding(.top, 16)

                    VStack(alignment: .leading, spacing: 8) {
                        MonoLabel(text: "DELIVERY IN")
                        HStack(spacing: 7) {
                            ForEach(Array(Seed.dayLabels.enumerated()), id: \.offset) { i, d in
                                Chip(title: d, on: s.quoteDayIdx == i) { store.setQuoteDay(i) }
                            }
                        }
                    }.padding(.top, 16)

                    VStack(alignment: .leading, spacing: 8) {
                        MonoLabel(text: "MESSAGE")
                        FKTextArea(placeholder: "What’s included, how you work, what you need from them.",
                                   text: Binding(get: { s.quoteMsg }, set: { store.setQuoteMsg($0) }), minHeight: 66)
                        if s.quoteError {
                            Text("Add a price and a short message before sending.").font(.system(size: 11.5, weight: .medium)).foregroundStyle(FK.alert)
                        }
                    }.padding(.top, 16)

                    PrimaryButton(title: "Send quote") { store.sendQuote() }.padding(.top, 18)
                }
                .padding(.horizontal, 20).padding(.top, 18).padding(.bottom, 22)
            }
        }
    }
}

File: android/app/src/main/java/com/freelancekar/app/ui/sheets/QuoteSheet.kt
package com.freelancekar.app.ui.sheets

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.text.KeyboardOptions
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.ModalBottomSheet
import androidx.compose.material3.Text
import androidx.compose.material3.TextField
import androidx.compose.material3.TextFieldDefaults
import androidx.compose.material3.rememberModalBottomSheetState
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.input.KeyboardType
import androidx.compose.ui.unit.dp
import com.freelancekar.app.data.AppState
import com.freelancekar.app.data.DAY_LABELS
import com.freelancekar.app.ui.AppViewModel
import com.freelancekar.app.ui.components.Chip
import com.freelancekar.app.ui.components.PrimaryButton
import com.freelancekar.app.ui.theme.FkColor
import com.freelancekar.app.ui.theme.FkType

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun QuoteSheet(state: AppState, vm: AppViewModel) {
    val req = state.reqs.find { it.id == state.quoteFor }
    if (req == null) return
    val sheetState = rememberModalBottomSheetState(skipPartiallyExpanded = true)

    ModalBottomSheet(
        onDismissRequest = { vm.closeQuote() },
        sheetState = sheetState,
        containerColor = FkColor.sheet,
    ) {
        Column(Modifier.fillMaxWidth().padding(horizontal = 20.dp).padding(bottom = 22.dp)) {
            Row(Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween) {
                Text("Send a quote", style = FkType.sheetTitle)
                Text("✕", style = FkType.body.copy(color = FkColor.text50), modifier = Modifier.clickable { vm.closeQuote() })
            }

            Column(
                Modifier
                    .fillMaxWidth()
                    .padding(top = 14.dp)
                    .clip(RoundedCornerShape(14.dp))
                    .background(FkColor.nested)
                    .border(1.dp, FkColor.line09, RoundedCornerShape(14.dp))
                    .padding(13.dp),
            ) {
                Text(req.title, style = FkType.cardTitle)
                Spacer(Modifier.height(4.dp))
                Text(req.meta, style = FkType.meta.copy(color = FkColor.text50))
                Spacer(Modifier.height(6.dp))
                Text("Client budget: ${req.budget}", style = FkType.meta.copy(color = FkColor.lime, fontWeight = androidx.compose.ui.text.font.FontWeight.Medium))
            }

            Column(Modifier.padding(top = 16.dp), verticalArrangement = Arrangement.spacedBy(8.dp)) {
                Text("YOUR PRICE", style = FkType.monoLabel.copy(color = FkColor.text45))
                Row(
                    Modifier
                        .fillMaxWidth()
                        .clip(RoundedCornerShape(13.dp))
                        .background(FkColor.nested)
                        .border(1.dp, FkColor.line09, RoundedCornerShape(13.dp))
                        .padding(horizontal = 13.dp, vertical = 2.dp),
                    verticalAlignment = Alignment.CenterVertically,
                ) {
                    Text("₹", style = FkType.body.copy(color = FkColor.text50))
                    Spacer(Modifier.width(9.dp))
                    TextField(
                        value = state.quoteAmount,
                        onValueChange = { vm.setQuoteAmount(it) },
                        placeholder = { Text("14000", style = FkType.body.copy(color = FkColor.text42)) },
                        modifier = Modifier.weight(1f),
                        singleLine = true,
                        colors = TextFieldDefaults.colors(
                            focusedContainerColor = FkColor.nested, unfocusedContainerColor = FkColor.nested,
                            focusedIndicatorColor = Color.Transparent, unfocusedIndicatorColor = Color.Transparent,
                        ),
                        textStyle = FkType.body.copy(color = FkColor.text, fontWeight = androidx.compose.ui.text.font.FontWeight.SemiBold),
                        keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Number),
                    )
                }
                Text("You keep 100% — client pays you directly, no platform cut.", style = FkType.meta.copy(color = FkColor.text42))
            }

            Column(Modifier.padding(top = 16.dp), verticalArrangement = Arrangement.spacedBy(8.dp)) {
                Text("DELIVERY IN", style = FkType.monoLabel.copy(color = FkColor.text45))
                Row(horizontalArrangement = Arrangement.spacedBy(7.dp)) {
                    DAY_LABELS.forEachIndexed { i, d ->
                        Box(Modifier.weight(1f)) {
                            Chip(on = state.quoteDayIdx == i, onClick = { vm.setQuoteDay(i) }, text = d, modifier = Modifier.fillMaxWidth())
                        }
                    }
                }
            }

            Column(Modifier.padding(top = 16.dp), verticalArrangement = Arrangement.spacedBy(8.dp)) {
                Text("MESSAGE", style = FkType.monoLabel.copy(color = FkColor.text45))
                TextField(
                    value = state.quoteMsg,
                    onValueChange = { vm.setQuoteMsg(it) },
                    placeholder = { Text("What's included, how you work, what you need from them.", style = FkType.body.copy(color = FkColor.text42)) },
                    modifier = Modifier
                        .fillMaxWidth()
                        .heightIn(min = 66.dp)
                        .clip(RoundedCornerShape(13.dp))
                        .border(1.dp, FkColor.line09, RoundedCornerShape(13.dp)),
                    colors = TextFieldDefaults.colors(
                        focusedContainerColor = FkColor.nested, unfocusedContainerColor = FkColor.nested,
                        focusedIndicatorColor = Color.Transparent, unfocusedIndicatorColor = Color.Transparent,
                    ),
                    textStyle = FkType.body.copy(color = FkColor.text),
                )
                if (state.quoteError) {
                    Text("Add a price and a short message before sending.", style = FkType.meta.copy(color = FkColor.alert))
                }
            }

            PrimaryButton({ vm.sendQuote() }, modifier = Modifier.padding(top = 18.dp), text = "Send quote")
        }
    }
}
